api/app/utils/redapp.py

```diff
@@ -0,0 +1,316 @@
+
+    comboIndexToFuel.put(Integer.valueOf(0), Integer.valueOf(0));
+    comboIndexToFuel.put(Integer.valueOf(1), Integer.valueOf(1));
+    comboIndexToFuel.put(Integer.valueOf(2), Integer.valueOf(2));
+    comboIndexToFuel.put(Integer.valueOf(3), Integer.valueOf(3));
+    comboIndexToFuel.put(Integer.valueOf(4), Integer.valueOf(4));
+    comboIndexToFuel.put(Integer.valueOf(5), Integer.valueOf(5));
+    comboIndexToFuel.put(Integer.valueOf(6), Integer.valueOf(6));
+    comboIndexToFuel.put(Integer.valueOf(7), Integer.valueOf(7));
+    comboIndexToFuel.put(Integer.valueOf(8), Integer.valueOf(8));
+    comboIndexToFuel.put(Integer.valueOf(9), Integer.valueOf(10));
+    comboIndexToFuel.put(Integer.valueOf(10), Integer.valueOf(11));
+    comboIndexToFuel.put(Integer.valueOf(11), Integer.valueOf(13));
+    comboIndexToFuel.put(Integer.valueOf(12), Integer.valueOf(14));
+    comboIndexToFuel.put(Integer.valueOf(13), Integer.valueOf(16));
+    comboIndexToFuel.put(Integer.valueOf(14), Integer.valueOf(17));
+    comboIndexToFuel.put(Integer.valueOf(15), Integer.valueOf(18));
+    comboIndexToFuel.put(Integer.valueOf(16), Integer.valueOf(19));
+    comboIndexToFuel.put(Integer.valueOf(17), Integer.valueOf(20));
+    comboIndexToFuel.put(Integer.valueOf(18), Integer.valueOf(21));
+
+    combobox index -> fuel type index
+    0:0 "C-1:  Spruce-Lichen Woodland"
+    1:1 "C-2:  Boreal Spruce"
+    2:2 "C-3:  Mature Jack or Lodgepole Pine"
+    3:3 "C-4:  Immature Jack or Lodgepole Pine"
+    4:4 "C-5:  Red and White Pine"
+    5:5 "C-6:  Conifer Plantation"
+    6:6 "C-7:  Ponderosa Pine / Douglas Fir"
+    7:7 "D-1:  Leafless Aspen"
+    8:8 "D-2:  Green Aspen (w/ BUI Thresholding)"
+    9:10 "D-1/D-2:  Aspen"
+    10:11 "M-1:  Boreal Mixedwood - Leafless"
+    11:13 "M-2:  Boreal Mixedwood - Green"
+    12:14 "M-1/M-2:  Boreal Mixedwood"
+    13:16 "M-3:  Dead Balsam Fir / Mixedwood - Leafless"
+    14:17 "M-4:  Dead Balsam Fir / Mixedwood - Green"
+    15:18 "M-3/M-4:  Dead Balsam Fir / Mixedwood"
+    16:19 "O-1a:  Matted Grass"
+    17:20 "O-1b:  Standing Grass"
+    18:21 "O-1ab:  Grass"
+    19"S-1:  Jack or Lodgepole Pine Slash"
+    20"S-2:  White Spruce / Balsam Slash"
+    21"S-3:  Coastal Cedar / Hemlock / Douglas-Fir Slash"
+    """
+    fuel_type_map = {
+        'C1': 0,
+        'C2': 1,
+        'C3': 2,
+        'C4': 3,
+        'C5': 4,
+        'C6': 5,
+        'C7': 6,
+        'D1': 7,
+        'D2': 8,
+        # D1/D2: Aspen # we don't have this as a dropdown right now!
+        'M1': 11,
+        'M2': 13,
+        # M1/M2: Boreal Mixedwood # we don't have this as a dropdown right now!
+        'M3': 16,
+        'M4': 17,
+        # M3/M4: Dead Balsam Fir / Mixedwood # we don't have this as a dropdown right now!
+        'O1A': 19,
+        'O1B': 20,
+        # O1ab: Grass # we don't have this as a dropdown right now!
+        # 'S1': - no mapping
+        # 'S2': - no mapping
+        # 'S3': - no mapping
```

Comment:
does this mean that diurnal FFMC can't be calculated for our S1-S3 fuel types?

Reply:
Oh - that mapping only applies to FBPCalculations - or to be exact - ca.redapp.ui.FbpTab - I'll update function name to reflect that.